cumulus_library_opioid/vocab/rxnorm_vsac_builder.py

```diff
@@ -19,65 +19,90 @@ def prepare_queries(
         **kwargs,
     ):
         def get_create_view_filter_by(
-            steward,
-            a_table,
-            b_table = None,
-            a_join_col = None,
-            b_join_col = None,
-            a_schema = None,
-            view_name = None,
-            **kwargs,
+            steward: str,
+            a_table: str,
+            *,
+            columns : list,
+            b_table:str | None= None,
+            a_join_col: str | None= None,
+            b_join_col: str | None= None,
+            a_schema: str | None= None,
+            view_name:str | None= None,
+            join_clauses: list | None = None,
+            column_aliases: dict | None = None
+            
         ):
             a_schema = a_schema or 'rxnorm.'
-            return base_templates.get_base_template(
-                'create_view_from_tables',
-                self.base_path / "template_sql",
+            return base_templates.get_create_view_from_tables(
                 view_name=view_name or (
                     f'{manifest.get_study_prefix()}__{steward}_{a_table}'
                 ),
-                a_table = f'{a_schema}{a_table}',
-                a_schema = a_schema,
-                b_table= b_table or f'opioid__{steward}_vsac',
-                a_join_col= a_join_col or 'rxcui',
-                b_join_col = b_join_col or 'code',
-                **kwargs
+                tables = [
+                    f'{a_schema}{a_table}',
+                    b_table or f'opioid__{steward}_vsac',
+                ],
+                table_aliases = ['a','b'],
+                join_clauses=join_clauses or [f"{a_join_col or 'a.rxcui'} = {b_join_col or 'b.code'}"],
```

Comment:
This line feels a bit awkward to me. Maybe move the arg-defaulting logic up to the top of the method? Like:

```
a_join_col = a_join_col or 'a.rxcui'
b_join_col = b_join_col or 'b.code'
join_clauses = join_clauses or [f"{a_join_col} = {b_join_col}"]
... 
get_create_view_from_tables(
  ...
  join_clauses=join_clauses,
)
```

Which feels.... slightly better

Reply:
can do.